pyvista/plotting/mapper.py

```diff
@@ -713,7 +712,6 @@ def set_scalars(
                     self.lookup_table.apply_cmap(cmap, n_colors)
                 else:  # pragma: no cover
                     # should be impossible to get to this point as VTK package
-                    # no requires matplotlib
```

Comment:
I think this was a typo, and the second part of this sentence was meant to be
```
# now requires matplotlib
```
In which case we should restore this line and fix the typo. Or reword the two-line comment to mention our own matplotlib dependency. Something like
```py
# should be impossible to get to this point as matplotlib is a dependency
```
(give or take line lengths and wording).

Or perhaps the whole comment is meaningless now, I haven't checked the big picture. In this case I'd raise in the `else`, maybe an `AssertionError` to signal that this should never happen.

Reply:
Turns out you can trigger this section of code even with `matplotlib` installed with:

```py
import pyvista

pyvista.global_theme.cmap = None

mesh = pyvista.Sphere()
mesh.plot(scalars=range(mesh.n_points))
```

This is because by setting `pyvista.global_theme.cmap = None`, now it's possible for `cmap` to be `None` within the mapper.

The easiest fix is to make sure that our theme must have a valid default color map. Since we now require `matplotlib`, we can do this and completely remove this section of code.